(BEDS-1338) moved premium perk ebs to vdb overview

remoterami opened the pull request.

remoterami committed:
refactor(api): moved premium perk ebs to vdb overview

sasha-bitfly commented:
@remoterami is broken frontend check addressed by Manuel?
![image](https://github.com/user-attachments/assets/ec5f31ee-baa9-4271-9f9d-8e13567964e7)

sasha-bitfly commented on backend/pkg/api/data_access/vdb_helpers.go:
```diff
@@ -410,16 +412,41 @@ func (d *DataAccessService) calculateValidatorDashboardBalance(ctx context.Conte
-			return balances, fmt.Errorf("error retrieving total cl deposits for non-rocketpool validators: %w", err)
+			return err
+		}
+		balances.EffectiveLatest = balances.EffectiveCurrent
```
balances.EffectiveLatest and balances.StakedEth are modified in two separate goroutines, a data race may potentially occur if balances.EffectiveLatest or balances.StakedEth become shared across multiple goroutines.

remoterami commented:
> @remoterami is broken frontend check addressed by Manuel?

Yes we're synchronizing the breaking API changes with all consumers :+1:

enzo-bitfly committed:
fix: display latest effective balance to unauthenticated users

- this adapts to API change that sends `latest effective balance` to a public endpoint

See: BEDS-1353
See: BEDS-1337

enzo-bitfly committed:
refactor: add prop type to set `displayCurrencyDefault.main` as `target-currency` in `BcFormatAmount.vue`

enzo-bitfly commented:
Blocked by #1404
